neutron_fwaas/services/firewall/fwaas_plugin_v2.py

```diff
@@ -162,12 +163,19 @@ def _validate_ports_for_firewall_group(self, context, tenant_id,
         # TODO(sridar): elevated context and do we want to use public ?
         for port_id in fwg_ports:
             port = self._core_plugin.get_port(context, port_id)
-
-            if port['tenant_id'] != tenant_id:
+            device_owner = port.get('device_owner', '')
+            port_tenant_id = port['tenant_id']
+            if device_owner == nl_constants.DEVICE_OWNER_ROUTER_GW:
+                # The gateway interface has no tenant, test if the gateway interface's router
```

Comment:
[optional] Line longer than 79 characters. Our coding standards (for our own drivers) generally use 120 chars, but OpenStack projects use 79. With our fork of neutron we use 79 as well. Guess this is where you'll have to make a decision how to handle this project :)

Reply:
I won't go with the 80 char line length as I feel that ll the line breaks make the code much less readable.